packages/core/admin/server/services/api-token.js

```diff
@@ -74,20 +74,13 @@ const create = async attributes => {
 /**
  * @returns {void}
  */
-const createSaltIfNotDefined = () => {
-  if (strapi.config.get('admin.apiToken.salt')) {
-    return;
-  }
-
-  if (process.env.API_TOKEN_SALT) {
+const checkSaltIsDefined = () => {
+  if (!strapi.config.get('admin.apiToken.salt')) {
+    const secretExample = crypto.randomBytes(16).toString('base64');
     throw new Error(
-      `There's something wrong with the configuration of your api-token salt. If you have changed the env variable used in the configuration file, please verify that you have created and set the variable in your .env file.`
+      `Missing admin.apiToken.salt. Please set admin.apiToken.salt in config/admin.js (ex: ${secretExample})`
```

Comment:
Since the value in `admin.apiToken.salt` defaults to the env variable, would it be more covenient to mention it here and add a link to the documentation for the admin configuration? This could also be a good place to explain what a `salt` actually is, since that might not be obvious.

Reply:
Actually the idea for me is to prevent the use of `API_TOKEN_SALT` as it is conflicting with the fact that you can set it in the config file too. The idea is to only use the config file.
I think I will propose to delete completely `API_TOKEN_SALT` (it's a breaking change, so we have to discuss it).

The recommended way to do would be that the user sets their own env var that they read in the config file :
```js
// config/admin.js
module.exports = ({ env }) => ({
  auth: {
    secret: env('ADMIN_JWT_SECRET'),
  },
  apiToken: {
    salt: env('API_TOKEN_SALT'),
  },
});

```